### PR TITLE
Issue #338 - Refactor RightclickManageServiceImpl

### PR DESCRIPTION
**Pull request by Giancarmine**

### CODE REVIEW

File: blockball-core/src/main/java/com/github/shynixn/blockball/core/logic/business/service/RightclickManageServiceImpl.kt [hunk 1]
```diff
@@ -35,14 +34,14 @@ import org.bukkit.entity.Player
...
-    private val rightClickListener = HashMap<Player, (Any) -> Unit>()
+class RightclickManageServiceImpl @Inject constructor(private val proxyService: ProxyService) : RightclickManageService {
+    private val rightClickListener = HashMap<Any, (Any) -> Unit>()
 
     /**
      * Gets called one time when a location gets rightlicked by [player].
      */
     override fun <P, L> watchForNextRightClickSign(player: P, f: (L) -> Unit) {
-        if (player !is Player) {
+        if (player !is Any) {
```
Comment (Shynixn) [inline blockball-core/src/main/java/com/github/shynixn/blockball/core/logic/business/service/RightclickManageServiceImpl.kt:44]: All of those (player !is Any) should be replaced by ``require(player is Any)``.

File: blockball-core/src/main/java/com/github/shynixn/blockball/core/logic/business/service/RightclickManageServiceImpl.kt [hunk 2]
```diff
@@ -54,19 +53,16 @@ class RightclickManageServiceImpl : RightclickManageService {
...
         if (!rightClickListener.containsKey(player)) {
             return false
         }
 
-        if ((location as Location).block.state is Sign) {
-            val sign = location.block.state as Sign
-            sign.setLine(0, ChatColor.BOLD.toString() + "BlockBall")
-            sign.setLine(1, ChatColor.GREEN.toString() + "Loading...")
-            sign.update(true)
+        if (location is Any) {
```
Comment (Shynixn) [inline blockball-core/src/main/java/com/github/shynixn/blockball/core/logic/business/service/RightclickManageServiceImpl.kt:64]: This if is pointless. 
Put proxyService.setSignLines outside of the if. Only if the result of proxyService.setSignLines is true, then call the rightClickListener statements.

**Review by Shynixn — CHANGES_REQUESTED** [review]
Looks good. Only some minor changes are required.

File: blockball-core/src/main/java/com/github/shynixn/blockball/core/logic/business/service/RightclickManageServiceImpl.kt (hunk 2 above)
Comment (Giancarmine) [reply blockball-core/src/main/java/com/github/shynixn/blockball/core/logic/business/service/RightclickManageServiceImpl.kt]: Check please!

**Review by Shynixn — APPROVED** [review]
That's ok for me. 

Thank you for your help! 👍

**Pull request merged**